### ISSUE TITLE
Errors on replication page of master (Warning in ./libraries/dbi/DBIMysqli.class.php#298)

### ISSUE THREAD

**Issue by KillerKelvUK** [phpmyadmin/phpmyadmin#13117]
### Steps to reproduce
1.  Configured master-slave replication between a mysql 5.7 master running in ubuntu 16.04 and a MariaDB 10.0.28 slave running on my Asustor NAS AS204-TE.
2.  Tested and confirmed replication is working and the slave has report_host configured.
3.  Open phpMyAdmin for the master to the Replication page and 2 error messages present.

Master:
phpmyadmin ver 4.5.4.1deb2ubuntu2. 
Apache/2.4.18 (Ubuntu)
PHP version: 7.0.15-0ubuntu0.16.04.4

Slave:
phpmyadmin ver 4.0.1
Apache/2.2.31 (Unix) DAV/2 PHP/5.6.14 mod_ssl/2.2.31 OpenSSL/1.0.2j

After ignoring the errors the 'Show master status' and 'Add slave replication user' links work as expected but the 'Show connected slaves' link returns no data.  However on the master...
`mysql> show slave hosts;`
+-----------+----------+------+-----------+------------+
| Server_id | Host     | Port | Master_id | Slave_UUID |
+-----------+----------+------+-----------+------------+
|         2 | blacknas | 3306 |         1 |            |
+-----------+----------+------+-----------+------------+
1 row in set (0.00 sec)

...which is what I'm expecting phpmyadmin to show a summary off?

### Actual behaviour
Error 1:
```
Warning in ./libraries/dbi/DBIMysqli.class.php#298
 mysqli_fetch_array(): Malformed server packet. Field length pointing 0 bytes after end of packet

Backtrace

./libraries/dbi/DBIMysqli.class.php#298: mysqli_fetch_array(
object,
integer 1,
)
./libraries/DatabaseInterface.class.php#2760: PMA_DBI_Mysqli->fetchAssoc(object)
./libraries/DatabaseInterface.class.php#2069: PMA_DatabaseInterface->fetchAssoc(object)
./libraries/replication_gui.lib.php#612: PMA_DatabaseInterface->fetchResult(
string 'SHOW SLAVE HOSTS',
NULL,
NULL,
)
./libraries/replication_gui.lib.php#56: PMA_getHtmlForReplicationSlavesTable(boolean true)
./server_replication.php#56: PMA_getHtmlForMasterReplication()
```

Error 2:
```
Warning in ./libraries/dbi/DBIMysqli.class.php#298
 mysqli_fetch_array(): Error while reading a row

Backtrace

./libraries/dbi/DBIMysqli.class.php#298: mysqli_fetch_array(
object,
integer 1,
)
./libraries/DatabaseInterface.class.php#2760: PMA_DBI_Mysqli->fetchAssoc(object)
./libraries/DatabaseInterface.class.php#2069: PMA_DatabaseInterface->fetchAssoc(object)
./libraries/replication_gui.lib.php#612: PMA_DatabaseInterface->fetchResult(
string 'SHOW SLAVE HOSTS',
NULL,
NULL,
)
./libraries/replication_gui.lib.php#56: PMA_getHtmlForReplicationSlavesTable(boolean true)
./server_replication.php#56: PMA_getHtmlForMasterReplication()
```

**Comment by nijel** [comment]
This really looks more like a bug in the PHP MySQL library than in our code. Can you please try to execute following PHP code to see if that works? You need to fill in username, password and maybe server.

```php
<?php
error_reporting(E_ALL);

$link = mysqli_connect("localhost", "my_user", "my_password");

/* check connection */
if (mysqli_connect_errno()) {
    printf("Connect failed: %s\n", mysqli_connect_error());
    exit();
}

$query = "SHOW SLAVE HOSTS";
$result = mysqli_query($link, $query);

/* get associative array */
while (($row = mysqli_fetch_array($result, MYSQLI_ASSOC)) !== null) {
    print_r($row)
}

/* free result set */
mysqli_free_result($result);

/* close connection */
mysqli_close($link);
?>

**Comment by KillerKelvUK** [comment]
Hi nijel, thanks for the help.  I had to put a ; at the end of row 17 to get your code to work, is that right as I'm not a coder?

Output:
```
PHP Warning:  mysqli_fetch_array(): Malformed server packet. Field length pointing 0 bytes after end of packet in /home/xxx/mysql_test on line 16
PHP Warning:  mysqli_fetch_array(): Error while reading a row in /home/xxx/mysql_test on line 16
Array
(
    [Server_id] => 2
    [Host] => blacknas
    [Port] => 3306
    [Master_id] => 1
)
```

**Comment by nijel** [comment]
Yes, sorry for that (I've fixed the code now as well).

So obviously it's PHP problem not related to phpMyAdmin. Probably the mysqlnd driver not correctly supporting something what MySQL server sends.

Quickly going through PHP bug tracker I've found https://bugs.php.net/bug.php?id=74021, but it seems different issue (and was actually fixed in version you're using). Anyway I suggest you to open new issue there.

Closing the issue here as it's something we certainly can not fix.
